tooling/nargo/src/ops/fuzz.rs

```diff
@@ -0,0 +1,349 @@
+                };
+                let mut fuzzer = FuzzedExecutor::new(
+                    acir_and_brillig_programs,
+                    acir_executor,
+                    brillig_executor,
+                    &package_name.clone().unwrap(),
+                    context.def_interner.function_name(&fuzzing_harness.get_id()),
+                    FuzzedExecutorExecutionConfiguration {
+                        num_threads: fuzz_execution_config.num_threads,
+                        timeout: fuzz_execution_config.timeout,
+                    },
+                    FuzzedExecutorFailureConfiguration {
+                        fail_on_specific_asserts: fuzzing_harness.only_fail_enabled(),
+                        failure_reason: fuzzing_harness.failure_reason(),
+                    },
+                    FuzzedExecutorFolderConfiguration {
+                        corpus_dir: fuzz_folder_config.corpus_dir.clone(),
+                        minimized_corpus_dir: fuzz_folder_config.minimized_corpus_dir.clone(),
+                    },
+                );
+
+                let result = fuzzer.fuzz();
+                match result {
+                    FuzzTestResult::Success => FuzzingRunStatus::ExecutionPass,
+                    FuzzTestResult::ProgramFailure(program_failure_result) => {
+                        // Collect failing callstack
+                        let unwrapped_acir_program = acir_program_copy.unwrap();
+                        let initial_witness = unwrapped_acir_program
+                            .abi
+                            .encode(&program_failure_result.counterexample.clone(), None)
+                            .unwrap();
+                        #[cfg(feature = "rpc")]
+                        let build_foreign_call_executor = |output, base| {
+                            DefaultForeignCallBuilder {
+                                output,
+                                enable_mocks: true,
+                                resolver_url: foreign_call_resolver_url.map(|s| s.to_string()),
+                                root_path: root_path.clone(),
+                                package_name: package_name.clone(),
+                            }
+                            .build_with_base(base)
+                        };
+                        #[cfg(not(feature = "rpc"))]
+                        let build_foreign_call_executor = |output, base| {
+                            DefaultForeignCallBuilder { output, enable_mocks: true }
+                                .build_with_base(base)
+                        };
+                        // Use a base layer that doesn't handle anything, which we handle in the `execute` below.
+                        let inner_executor = build_foreign_call_executor(
+                            if show_output { PrintOutput::Stdout } else { PrintOutput::None },
+                            layers::Unhandled,
+                        );
```

Comment:
This too could be deduplicated using that helper function.

Reply:
Done